python/instrumentation/openinference-instrumentation-langchain/src/openinference/instrumentation/langchain/__init__.py

```diff
@@ -64,6 +64,26 @@ def _uninstrument(self, **kwargs: Any) -> None:
     def get_span(self, run_id: UUID) -> Optional[Span]:
         return self._tracer.get_span(run_id) if self._tracer else None
 
+    def get_ancestors(self, run_id: UUID) -> Optional[List[Span]]:
+        ancestors = []
+        tracer = self._tracer
+        assert tracer
+
+        run = tracer.run_map.get(str(run_id))
+        if not run:
+            return None
+
+        run_id = run.parent_run_id  # start with the first ancestor
+
+        while run_id:
+            span = self.get_span(run_id)
+            if span:
+                ancestors.append(span)
+
+            run = tracer.run_map.get(str(run_id))
+            run_id = run.parent_run_id
+        return ancestors if ancestors else None
```

Comment:
I think for the simplicity of the typing (and maybe just my FP brain) - I think an empty array might be preferable to represent "none"?